third-party/realdds/include/realdds/dds-stream-base.h

```diff
@@ -0,0 +1,55 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+
+#include "dds-stream-profile.h"
+
+#include <memory>
+#include <string>
+
+
+namespace realdds {
+
+
+class dds_topic;
+
+
+// Base class for both client/subscriber and server/publisher stream implementations: contains
+// information needed to identify a stream, its properties, and its profiles.
+//
+class dds_stream_base : public std::enable_shared_from_this< dds_stream_base >
+{
+protected:
+    std::string const _name;
+    std::string const _sensor_name;
+    int _default_profile_index = 0;
+    dds_stream_profiles _profiles;
+
+    dds_stream_base( std::string const & stream_name, std::string const & sensor_name );
+    
+public:
+    virtual ~dds_stream_base() = default;
+
+    // This function can only be called once!
+    void set_profiles( dds_stream_profiles const & profiles, int default_profile_index = 0 );
```

Comment:
Using `set_profiles` is not generic enough for our topics.
If we will want to send the profiles in several messages, not all profiles of a stream in a single message, than we will want to add them to the stream object upon receiving the message.
Why not call it `add_profiles` and allow it to be called several times? Maybe up to a maximum number of profiles set at construction time

Reply:
I don't understand the part about generic-enough.
Why would we want to send profiles in several messages?
You build up the definition of a stream and only then create it. I don't like incrementally adding profiles -- a stream is not something that's built over time, or changeable -- it's set by the server and that's it.

Reply:
We might want to send smaller topics, we still don't know HKR limitations.
There are some cameras with 350+ profiles for a given stream. Depending on stream and profile information we might get a very large message that we will want to break into several smaller messages.

But I can accept that we will collect all the information in the client and initialize only after receiving it.